file-formats/enhs/type/zif_seef_aff_enhs_v1.intf.abap

```diff
@@ -0,0 +1,223 @@
+
+  TYPES:
+    "! <p class="shorttext">Value Check Class</p>
+    "! Value check via class
+    BEGIN OF ty_value_check_class,
+      "! <p class="shorttext">Name of the Value Check Class</p>
+      "! Name of the value check class
+      "! $required
+      object_name    TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Length of the Value Check</p>
+      "! Length of the value check
+      length         TYPE i,
+      "! <p class="shorttext">Decimal Places of the Value Check</p>
+      "! Decimal places of the value check
+      decimal_places TYPE i,
+    END OF ty_value_check_class.
+
+  TYPES:
+    "! <p class="shorttext">Filter</p>
+    "! Information about this filter
+    BEGIN OF ty_badi_def_filter,
+      "! <p class="shorttext">Name of the Filter</p>
+      "! Name of the filter
+      "! $required
+      name                 TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Description of the Filter</p>
+      "! Description of the filter
+      "! $required
+      description          TYPE c LENGTH 255,
+      "! <p class="shorttext">Type of the Filter</p>
+      "! Type of the filter
+      "! $required
+      type                 TYPE ty_filter_type,
+      "! <p class="shorttext">Only constant Filter Values</p>
+      "! The filter has only constant filter values
+      only_constant_values TYPE abap_bool,
+      "! <p class="shorttext">Value Check Ddic</p>
+      "! Value check via domain or data element
+      value_check_ddic     TYPE ty_value_check_ddic,
+      "! <p class="shorttext">Value Check Class</p>
+      "! Value check via class
+      value_check_class    TYPE ty_value_check_class,
+    END OF ty_badi_def_filter.
+
+  TYPES ty_badi_def_filters TYPE STANDARD TABLE OF ty_badi_def_filter WITH DEFAULT KEY.
+  TYPES ty_example_classes TYPE STANDARD TABLE OF zif_aff_types_v1=>ty_object_name_30 WITH DEFAULT KEY.
+
+
+  TYPES:
+    "! <p class="shorttext">BAdI Definition</p>
+    "! Information about this BAdI definition
+    BEGIN OF ty_badi_def,
+      "! <p class="shorttext">Name of the BAdI Definition</p>
+      "! Name of the BAdI definition
+      "! $required
+      name               TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Description of the BAdI Definition</p>
+      "! Description of the BAdI definition
+      "! $required
+      description        TYPE c LENGTH 255,
+      "! <p class="shorttext">Interface of the BAdI Definition</p>
+      "! Interface of the BAdI definition
+      "! $required
+      interface          TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Instantiation of the BAdI Definition</p>
+      "! Instantiation of the BAdI definition
+      instantiation      TYPE ty_instantiation,
+      "! <p class="shorttext">Single- or Multiple Use BAdI</p>
+      "! Indicator whether the BAdI definition is single- or multiple usable
+      multiple_use       TYPE abap_bool,
```

Comment:
If we touch it now, maybe you want to check discussion in https://github.com/SAP/abap-file-formats/issues/61#issuecomment-839606098

Reply:
If we changed this to an enum (or rather our "constant enum type"), wouldn't it result in changing all booleans to make it consistent? Or is this a special case?

Reply:
Sorry, I missed to add a point from #61. It was about whether empty values shall be displayed. I took this example that we might want to show always whether a BAdI definition is configured as singleUsed or multipleUsed. 
I see three options here:

1. Only show the value, if it is a multipleUsed BAdI
2. We use `$showAlways` here
3. We change it to an enum

For me, all options work. BTW- adding or removing `$showAlways` can be considered as compatible change [here](https://github.com/SAP/abap-file-formats/blob/main/doc/specification.md#compatible-file-format-changes)

Reply:
Personally, I prefer the $showAlways tag. That's why we introduced it